### PR TITLE
Adjust end time, not duration in adjust_time

### PR DESCRIPTION
**Pull request by cifkao**
We want `func` to be applied to absolute timestamps, not durations.

### CODE REVIEW

**Comment by cifkao** [comment]
It's not easy to get this right...

It should be OK now. Should I try to fix the coverage? (We're not testing `Chord` because there are no chords in the test data.)

**Comment by salu133445** [comment]
Looks like a bug to me. Once `time` is adjusted, `end` (a property that returns `time + duration`) is no longer accurate.

However, `time` and `duration` are two different attributes, so we should handle them separately if possible for consistency reason. It can be quite tricky to adjust them independently due to potential accumulations of rounding errors.

**Comment by cifkao** [comment]
It doesn't seem to be possible to me to adjust them separately without caching the old time value somewhere... Or making sure duration gets adjusted first, which would be quite fragile.

I think maybe classes like `Note` and `Chord` should just override `adjust_time` instead of `_adjust_time`. That way the function will get called only once for the whole object and interactions between attributes can be handled easily.

**Comment by cifkao** [comment]
How about this?

**Comment by salu133445** [comment]
Merged. Thank you!

File: muspy/classes.py [hunk 1]
```diff
@@ -331,10 +331,38 @@ def _validate(self, attr: str, recursive: bool):
...
+        recursive : bool
+            Whether to apply recursively. Defaults to True.
+
+        Returns
+        -------
+        Object itself.
+
+        """
+        if attr is not None and attr != "time":
+            raise AttributeError(f"'Note' object has no attribute '{attr}'")
```
Comment (cifkao) [inline muspy/classes.py:360]: This might be confusing since there are other attributes than "time", we just don't support adjusting them.

Comment (salu133445) [reply muspy/classes.py:360]: Oops. You are right. I will fix this later.

In fact, several methods of Base do not handle `AttributeError` properly.

**Pull request merged**